kernel/src/syscall.rs

```diff
@@ -166,10 +167,12 @@ pub enum Syscall {
         driver_number: usize,
         /// The subscribe identifier.
         subdriver_number: usize,
+        // On CHERI platforms we need to maintain metadata for
+        // these two as they are used to access code/data.
```

Comment:
```suggestion
```

I think this is a useful comment for context in this PR, but given the "meaning" of `CapabilityPtr` I don't think it's useful to justify this in terms of CHERI